homeassistant/components/whirlpool/climate.py

```diff
@@ -0,0 +1,189 @@
+        return
+
+    aircon = AirConEntity(said_list[0], auth)
+    async_add_entities([aircon], True)
+
+
+class AirConEntity(ClimateEntity):
+    """Representation of an air conditioner."""
+
+    _attr_fan_modes = [FAN_AUTO, FAN_HIGH, FAN_MEDIUM, FAN_LOW, FAN_OFF]
+    _attr_hvac_modes = [
+        HVAC_MODE_COOL,
+        HVAC_MODE_HEAT,
+        HVAC_MODE_FAN_ONLY,
+        HVAC_MODE_OFF,
+    ]
+    _attr_max_temp = 30
+    _attr_min_temp = 16
+    _attr_supported_features = (
+        SUPPORT_TARGET_TEMPERATURE | SUPPORT_FAN_MODE | SUPPORT_SWING_MODE
+    )
+    _attr_swing_modes = [SWING_HORIZONTAL, SWING_OFF]
+    _attr_target_temperature_step = 1
+    _attr_temperature_unit = TEMP_CELSIUS
+
+    def __init__(self, said, auth: Auth):
+        """Initialize the entity."""
+        self._aircon = Aircon(auth, said, self.schedule_update_ha_state)
+
+        self._attr_name = said
+        self._attr_unique_id = said
+
+    async def async_added_to_hass(self) -> None:
+        """Connect aircon to the cloud."""
+        await self._aircon.connect()
+
+        try:
+            name = await self._aircon.fetch_name()
+            if name is not None:
+                self._attr_name = name
+        except (asyncio.TimeoutError, aiohttp.ClientError):
+            _LOGGER.exception("Failed to get name")
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._aircon.get_online()
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._aircon.get_current_temp()
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._aircon.get_temp()
+
+    async def async_set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        await self._aircon.set_temp(kwargs.get(ATTR_TEMPERATURE))
+
+    @property
+    def current_humidity(self):
+        """Return the current humidity."""
+        return self._aircon.get_current_humidity()
+
+    @property
+    def target_humidity(self):
+        """Return the humidity we try to reach."""
+        return self._aircon.get_humidity()
+
+    async def async_set_humidity(self, humidity: int) -> None:
+        """Set new target humidity."""
+        await self._aircon.set_humidity(humidity)
+
+    @property
+    def hvac_mode(self):
+        """Return current operation ie. heat, cool, fan."""
+        if not self._aircon.get_power_on():
+            return HVAC_MODE_OFF
+
+        mode: AirconMode = self._aircon.get_mode()
+        return AIRCON_MODE_MAP.get(mode, None)
+
+    async def async_set_hvac_mode(self, hvac_mode):
+        """Set HVAC mode."""
+        if hvac_mode == HVAC_MODE_OFF:
+            await self._aircon.set_power_on(False)
+            return
+
+        mode = HVAC_MODE_TO_AIRCON_MODE.get(hvac_mode)
+        if not mode:
+            _LOGGER.warning("Unexpected hvac mode")
```

Comment:
Should we log the `hvac_mode`?
```suggestion
            _LOGGER.warning("Unexpected hvac mode: %s", hvac_mode)
```